webapp/src/ts/modules/analytics/analytics.component.ts

```diff
@@ -1,49 +1,77 @@
-import {Component, OnInit} from '@angular/core';
-import { Store } from '@ngrx/store';
-import { AnalyticsModulesService } from '../../services/analytics-modules.service';
-import { AnalyticsActions } from '../../actions/analytics';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
-import * as _ from 'lodash-es';
+import { Store } from '@ngrx/store';
+import { find as _find } from 'lodash-es';
+import { combineLatest, Subscription } from 'rxjs';
+
+import { AnalyticsModulesService } from '@mm-services/analytics-modules.service';
+import { AnalyticsActions } from '@mm-actions/analytics';
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
 
 @Component({
   templateUrl: './analytics.component.html'
 })
-export class AnalyticsComponent implements OnInit {
+export class AnalyticsComponent implements OnInit, OnDestroy {
   private analyticsActions;
-
+  private globalActions;
+  subscriptions: Subscription = new Subscription();
   analyticsModules = [];
-  loading = true;
 
   constructor(
     private store: Store,
-    private analyticsModulesService:AnalyticsModulesService,
-    private route:ActivatedRoute,
-    private router:Router,
-    //private tour:Tour Todo
+    private analyticsModulesService: AnalyticsModulesService,
+    private route: ActivatedRoute,
+    private router: Router,
+    //private tour:Tour Todo: integrate once tour feature is complete
   ) {
     this.analyticsActions = new AnalyticsActions(store);
+    this.globalActions = new GlobalActions(store);
   }
 
   ngOnInit() {
-    // todo unset selected
+    this.subscribeToStore();
     this.analyticsActions.setSelectedAnalytics(null);
+    this.globalActions.unsetSelected();
+    this.getAnalyticsModules();
 
-    this.analyticsModulesService.get().then((modules) => {
-      if (this.route.snapshot.routeConfig.path === 'analytics') {
-        if (modules.length === 1) {
-          return this.router.navigate([modules[0].route]);
-        }
-      } else {
-        this.analyticsActions.setSelectedAnalytics(_.find(modules, { route: this.route.snapshot.routeConfig.path }));
-      }
+    /*
+    Todo: integrate once tour feature is complete
+    if ($stateParams.tour) {
+      Tour.start($stateParams.tour);
+    }
+    */
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.unsubscribe();
+  }
 
-      this.loading = false;
-      this.analyticsModules = modules;
+  private subscribeToStore() {
+    const selectorsSubscription = combineLatest(
+      this.store.select(Selectors.getAnalyticsModules),
+    )
+    .subscribe(([
+      analyticsModules = [],
+    ]) => {
+      this.analyticsModules = analyticsModules;
     });
+    this.subscriptions.add(selectorsSubscription);
+  }
 
-    /*if ($stateParams.tour) {
-      Tour.start($stateParams.tour);
-    }*/
+  getAnalyticsModules() {
```

Comment:
Can we make this private? In tests, you can just re-initialize the component to get this to execute.

Reply:
Fixed.